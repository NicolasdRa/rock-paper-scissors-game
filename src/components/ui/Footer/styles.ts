import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      position: "absolute",
      bottom: 0,
      right: "2rem",
      display: "flex",
      flexDirection: "row",
      justifyContent: "flex-end",
      alignItems: "center",

      [theme.breakpoints.down("md")]: {
        justifyContent: "space-between",
        right: "1.5rem",
      },
    },
  })
);
