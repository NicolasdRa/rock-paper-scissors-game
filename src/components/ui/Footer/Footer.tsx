import { Box } from "@material-ui/core";

import { RulesModal } from "../RulesModal/RulesModal";
import { StartBtn } from "../StartBtn/StartBtn";
import { useStyles } from "./styles";

export const Footer = () => {
  const classes = useStyles();

  return (
    <Box className={classes.footer}>
      <Box>
        <StartBtn />
      </Box>
      <Box>
        <RulesModal />
      </Box>
    </Box>
  );
};
