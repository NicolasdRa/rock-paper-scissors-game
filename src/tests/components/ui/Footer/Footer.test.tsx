import "@testing-library/jest-dom";
import { mount, ReactWrapper } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Footer } from "../../../../components/ui/Footer/Footer";

describe("tests on <Footer/>", () => {
  const wrapper = mount(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );

  test("should render <Footer/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
