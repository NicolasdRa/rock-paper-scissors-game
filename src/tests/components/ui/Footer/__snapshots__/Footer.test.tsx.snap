// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Footer/> should render <Footer/> correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-footer-1"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-startBtn-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sart New Match
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-5"
  >
    <div
      className="MuiBox-root MuiBox-root-14 makeStyles-main-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-btn-13 MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Rules
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
